components/raftstore-v2/src/operation/command/write/mod.rs

```diff
@@ -118,12 +118,14 @@ impl<EK: KvEngine, ER: RaftEngine> Peer<EK, ER> {
 
     pub fn propose_pending_writes<T>(&mut self, ctx: &mut StoreContext<EK, ER, T>) {
         if let Some(encoder) = self.simple_write_encoder_mut().take() {
+            let header = encoder.header();
+            let validate_res = self.validate_command(header, None, &mut ctx.raft_metrics);
             let call_proposed_on_success = if encoder.notify_proposed() {
                 // The request has pass conflict check and called all proposed callbacks.
```

Comment:
Here if notify_proposed() return false, then it would check the region epoch. The problem  is when encoder.notify_proposed() == true?
Why when encoder.notify_proposed() is true, it can bypass the epoch check, which apparently is not correct in this case.

Reply:
Now, the region epoch will be checked in validate_command anyway.